application/data_access/entity_queries.py

```diff
@@ -60,27 +60,29 @@ def get_entity_search(parameters: dict):
     params = normalised_params(parameters)
 
     with get_context_session() as session:
-        query = session.query(
-            EntityOrm, func.count(EntityOrm.entity).over().label("count")
-        )
+        only_fields = params.get("field")
+        if only_fields:
+            query_args = [getattr(EntityOrm, field.value) for field in only_fields]
+        else:
+            query_args = [EntityOrm]
+        query = session.query(*query_args)
         query = _apply_base_filters(query, params)
         query = _apply_date_filters(query, params)
         query = _apply_location_filters(session, query, params)
         query = _apply_entries_option_filter(query, params)
         query = _apply_limit_and_pagination_filters(query, params)
 
         entities = query.all()
+        count = query.count()
```

Comment:
That's a good point, i'll revert this and add in a test case to guard against it in the future

Reply:
An alternative would be to keep the current implementation, but to take the count before the pagination options are added; something like:

```
        query = _apply_entries_option_filter(query, params)
        count = query.count()
        query = _apply_limit_and_pagination_filters(query, params)
```

What do you think?

Reply:
I'd just add the window function back in so that line 68 would look like this:
query = session.query(*query_args, func.count(EntityOrm.entity).over().label("count"))

You'd just need to revert the assigning of the count value and the iteration of the resulting entity list.

Reply:
@Staberinde It's not a show stopper for me, but I don't see why you run two queries instead of just adding another arg to the session.query(*query_args) i.e. add the count to the query (func.count(EntityOrm.entity).over()). I'm not convinced it's more complex. I don't know which is more expensive, but having to do a separate count in a special place i.e. between apply all the other filters and applying the pagination if anything seems more complex. So if you do add it there, probably comment why it has to happen at that point. Otherwise merge away.

Reply:
The way I was doing it was certainly more expensive, as it involved the overhead associated with executing two queries instead of one. Thinking about it, another benefit of the inline method is that it'll probably give a more reliable result in the event we make these tables relational. Given that we're transforming the data to return in the `field` case anyway, I agree that it doesn't really make the code that much more complex in reality.